### PR TITLE
Removes deprecation warning.

### PR DESCRIPTION
**Pull request by rossjones**
There's no ticket for this because it's cleanup.

Tests were logging ```DeprecationWarning: BaseException.message has been deprecated as of Python 2.6``` every time something tried to extract a message from an exception extending
ActionError.  This PR fixes that and also does PEP8 on the __init__.py file.

### CODE REVIEW

**Comment by davidread** [comment]
What about instead we change occurrences of `e.messages` to `str(e)`? I assume that's more pythonic going forward?

**Comment by wardi** [comment]
is str(e) safe? what happens if e.messages was turned into a unicode object?  Just trying to avoid another #2596

**Comment by davidread** [comment]
There are only four places .message is done, according to the tests:
```
/home/travis/build/ckan/ckan/ckan/controllers/package.py:1249: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6
  error_message = (e.message or e.error_summary

/home/travis/build/ckan/ckan/ckan/controllers/package.py:1253: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6
  error_message = e.message

/home/travis/build/ckan/ckan/ckan/controllers/user.py:687: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6
  error_message = (e.message or e.error_summary

/home/travis/build/ckan/ckan/ckan/controllers/user.py:707: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6
  error_message = e.message
```
All of which just do:

    h.flash_error(error_message)

so I think calling unicode(e) would better and safer than overriding Exception.message.

**Comment by wardi** [comment]
@davidread :+1:

File: ckan/logic/__init__.py [hunk 1]
```diff
@@ -25,7 +25,14 @@ class UsernamePasswordError(Exception):
 
 
 class ActionError(Exception):
-    pass
+
+    def __init__(self, message=''):
+        self.message = message
+        super(ActionError, self).__init__(message)
+
+    def __str__(self):
```
Comment (wardi) [inline ckan/logic/__init__.py]: this should be `__unicode__` instead, for the cases where the message passed was a unicode object, right?

**Pull request merged**